src/window.service.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class WindowService {
+
+  get nativeWindow(): any {
+    return _window();
+  }
+}
+
+function _window(): any {
+  // return the global native browser window object
+  return typeof window !== 'undefined' ? window : undefined;
+}
```

Comment:
Yes you should wrap it in a function that falls back to global when window is undefined. 
but my point about implementing `WindowService` is to make the module work the angular way. I also recommend wrapping your code into a service like `LazyImageService`